src/Domain/CoordinateMaps/CoordinateMap.tpp

```diff
@@ -351,6 +351,142 @@ auto CoordinateMap<SourceFrame, TargetFrame, Maps...>::jacobian_impl(
+                               time, functions_of_time,
+                               domain::is_jacobian_time_dependent_t<Map, T>{});
+          for (size_t target = 0; target < dim; ++target) {
+            for (size_t source = 0; source < dim; ++source) {
+              jac.get(target, source) =
+                  std::move(noframe_jac.get(target, source));
+              inv_jac.get(source, target) =
+                  std::move(noframe_inv_jac.get(source, target));
+            }
+          }
+
+          // Set frame velocity
+          if (domain::is_map_time_dependent_v<
+                  std::tuple_element_t<0, std::decay_t<decltype(maps)>>>) {
+            std::array<T, dim> noframe_frame_velocity =
+                detail::get_frame_velocity(map, mapped_point, time,
+                                           functions_of_time);
+            for (size_t i = 0; i < dim; ++i) {
+              frame_velocity.get(i) = gsl::at(noframe_frame_velocity, i);
+            }
+          } else {
+            // If the first map is time-independent the velocity is initialized
+            // to zero
+            for (size_t i = 0; i < frame_velocity.size(); ++i) {
+              frame_velocity[i] = make_with_value<T>(get<0, 0>(jac), 0.0);
+            }
+          }
+        } else if (LIKELY(not map.is_identity())) {
+          // WARNING: we have assumed that if the map is the identity the frame
+          // velocity is also zero. That is, we do not optimize for the map
+          // being instantaneously zero.
+
+          detail::get_inv_jacobian(
+              make_not_null(&noframe_inv_jac), map, mapped_point, time,
+              functions_of_time,
+              domain::is_jacobian_time_dependent_t<Map, T>{});
+          detail::get_jacobian(make_not_null(&noframe_jac), map, mapped_point,
+                               time, functions_of_time,
+                               domain::is_jacobian_time_dependent_t<Map, T>{});
+
+          // Perform matrix multiplication for Jacobian and inverse Jacobian
+          detail::multiply_inv_jacobian(make_not_null(&inv_jac),
+                                        noframe_inv_jac);
+          detail::multiply_jacobian(make_not_null(&jac), noframe_jac);
+
+          // Set frame velocity, only if map is time-dependent
+          if (domain::is_map_time_dependent_v<
+                  std::tuple_element_t<count, std::decay_t<decltype(maps)>>>) {
+            std::array<T, dim> noframe_frame_velocity =
+                detail::get_frame_velocity(map, mapped_point, time,
+                                           functions_of_time);
+            for (size_t target_frame_index = 0; target_frame_index < dim;
+                 ++target_frame_index) {
+              for (size_t source_frame_index = 0; source_frame_index < dim;
+                   ++source_frame_index) {
+                frame_velocity.get(target_frame_index) +=
```

Comment:
I'm having trouble with the velocity logic. My understanding of this utility is that you're working through a composition of the maps in the tuple, and that here the `noframe_jac` is the jacobian for the current map being considered, while `jac` will be the cumulative jacobian for all of the maps, if the tuple contains {M_1, M_2, M_3}, this represents composition of maps M_3 * M_2 * M_1 and an overall jacobian J_3^i_j J_2^j_k J_1^k_i (I think that's consistent with your Jacobian math above)

The time dependent maps look to be giving the set of velocities dx^i/dt of the target coordinates. So a single map is obvious, the composition of two maps should give something like:
v_tot^i = v_2^i + J_2^i_j v_1^j
and three maps should be something like:
v_tot^i = v_3^i + J_3^i_j v_2^j + J_3^i_j J_2^j_k v_1^k

But here it looks like you are doing:
v_tot^i = v_1^i + J_2^i_j v_2^j + J_3^i_j v_3^j

What am I missing here?

Reply:
Nothing, what you have is exactly what I have in my notes, but it looks like I coded it wrong. I'm going to see if I can get a test that exposes the issue. I think a translation + affine map + translation might be enough, otherwise I think composing the cubic scale and translation should work.